src/test/scala/rna16sIdentification.scala

```diff
@@ -0,0 +1,164 @@
+    */
+  lazy val is16s: Entry => Dropped + Entry =
+    database
+      .filterIncluded(_)
+      .right
+      .flatMap(annotation.hasRNAType16s)
+      .right
+      .flatMap(annotation.someDescriptionContains16s)
+      .right
+      .flatMap(sequence.has16sLength)
+      .right
+      .flatMap(sequence.qualityOK)
+
+  /** Database filters */
+  case object database {
+
+    /** Databases from which we take sequences and annotations. */
+    lazy val includedDatabases: Set[Database] = {
+
+      import Database._
+
+      Set(
+        ENA,
+        RefSeq,
+        GreenGenes,
+        RDP,
+        Rfam
+      )
+    }
+
+    /** Keep only entry and sequence annotations coming from [[includedDatabases]].
+
+    The full [[Entry]] will be dropped if it doesn't have at least one annotation of both types (entry and sequence) coming from [[includedDatabases]].
+      */
+    lazy val filterIncluded: Entry => Dropped + Entry =
+      entry => {
+
+        val includedEntryAnnotations =
+          entry.entryAnnotations
+            .filter { a =>
+              includedDatabases contains a.databaseEntry.database
+            }
+
+        val includedSequenceAnnotations =
+          entry.sequenceAnnotations
+            .filter { sa =>
+              includedEntryAnnotations exists {
+                _.ncbiTaxonomyID == sa.ncbiTaxonomyID
+              }
+            }
+
+        if (includedEntryAnnotations.isEmpty || includedSequenceAnnotations.isEmpty)
+          Left(Dropped.FromExcludedDBs(entry))
+        else
+          Right(
+            entry.copy(
+              entryAnnotations = includedEntryAnnotations,
+              sequenceAnnotations = includedSequenceAnnotations
+            )
+          )
+      }
+  }
+
+  case object sequence {
+
+    /** Keep the [[Entry]] if it [[hasLength16s]]. */
+    lazy val has16sLength: Entry => Dropped + Entry =
+      entry =>
+        if (hasLength16s(entry))
+          Right(entry)
+        else
+          Left(
+            Dropped.InvalidSequenceLength(entry.rnaSequence.sequence.length,
+                                          entry))
+
+    /** true if it lies in the accepted 16S sequence length interval. */
+    lazy val length16s: Int => Boolean =
+      len => len >= 1300 && len <= 1700
+
+    /** checks if an [[Entry]] length is within the accepted 16S length interval. @see [[length16s]]*/
+    lazy val hasLength16s: Entry => Boolean =
+      entry => length16s(entry.rnaSequence.sequence.length)
+
+    /** Keep the [[Entry]] if the sequence
+
+        - has an acceptable percentage of `N`s [[ratioOfNs]]
+        - has no long `N...N` subsequences [[containsSliceOfNs]]
+      */
+    lazy val qualityOK: Entry => Dropped + Entry =
+      entry =>
+        if (ratioOfNs(entry.rnaSequence.sequence) <= 0.01 &&
+            !containsSliceOfNs(5)(entry.rnaSequence.sequence))
```

Comment:
Same question as above, out of simple curiosity: where do these numbers come from?

Reply:
Replied in https://github.com/ohnosequences/db.rna16s/pull/76#discussion_r167510673. Github UI is crap.

Reply:
Argh, sorry, didn't see it.